File: src/engine/commands/admin/addshieldcommand.cpp
#include "addshieldcommand.h"

#include "shield.h"


#define super AdminCommand

AddShieldCommand::AddShieldCommand(QObject *parent) :
    super(parent) {

    setDescription("Add a shield to the current room.\n"
                   "\n"
                   "Usage: add-shield <shield-name>");
}

AddShieldCommand::~AddShieldCommand() {
}

void AddShieldCommand::execute(Character *player, const QString &command) {

    super::prepareExecute(player, command);

    if (!assertWordsLeft("Usage: add-shield <shield-name>")) {
        return;
    }

    QString shieldName = takeWord();

    Shield *shield = new Shield(realm());
    shield->setName(shieldName);
    currentRoom()->addItem(shield);

    send("Shield %1 added.", shieldName);
}

File: data/commands/admin/addshieldcommand.js
function AddShieldCommand() {
    AdminCommand.call(this);

    this.setDescription("Add a shield to the current room.\n" +
                        "\n" +
                        "Usage: add-shield <shield-name>");
}

AddShieldCommand.prototype = new AdminCommand();
AddShieldCommand.prototype.constructor = AddShieldCommand;

AddShieldCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    if (!this.assertWordsLeft("Usage: add-shield <shield-name>")) {
        return;
    }

    var shieldName = this.takeWord();

    var shield = Realm.createObject("Shield");
    shield.name = shieldName;
    this.currentRoom.addItem(shield);

    this.send("Shield %1 added.", shieldName);
};

CommandRegistry.registerAdminCommand("add-shield", new AddShieldCommand());
